### PR TITLE
audio: Store sound transform values in 30-bit integers

### PR DESCRIPTION
**Pull request by relrelb**
It seems like Flash stores sound transform values in 30-bit unsigned integers:
* In general, negative values are equivalent to their absolute value.
* Specifically, 0x40000000, -0x40000000 and -0x80000000 are equivalent to zero.

Plus a small cleanup.

Fixes the "marching" sound in #5126, which roughly does the following:
```as
var s = new Sound();
s.attachSound("marching");
s.setPan(NaN);
s.start();
```
Previously this played a terrible noise in Ruffle, while it plays normally in Flash as if no transform is applied.

### CODE REVIEW

**Comment by midgleyc** [comment]
Looks good to me, one question: do both "x * x" and "x.pow(2)" compile down to the same thing for f32? If not, is there a behaviour / speed difference?

**Comment by adrian17** [comment]
It's actually an i32 multiplication, and only then converted to f32.
Either way, it's a `const`, so it should be calculated at compile time.

**Comment by midgleyc** [comment]
Yeah, looks like both forms just go to a constant. This looks good to me, then.

File: core/src/backend/audio.rs [hunk 1]
```diff
@@ -560,16 +557,19 @@ pub struct SoundTransform {
...
+        // * Specifically, 0x40000000, -0x40000000 and -0x80000000 are equivalent to zero.
         Self {
-            left_to_left: other.left_to_left as f32 * other.volume as f32 / SCALE,
-            left_to_right: other.left_to_right as f32 * other.volume as f32 / SCALE,
-            right_to_left: other.right_to_left as f32 * other.volume as f32 / SCALE,
-            right_to_right: other.right_to_right as f32 * other.volume as f32 / SCALE,
+            left_to_left: ((other.left_to_left << 2) >> 2) as f32 * other.volume as f32 / SCALE,
+            left_to_right: ((other.left_to_right << 2) >> 2) as f32 * other.volume as f32 / SCALE,
+            right_to_left: ((other.right_to_left << 2) >> 2) as f32 * other.volume as f32 / SCALE,
+            right_to_right: ((other.right_to_right << 2) >> 2) as f32 * other.volume as f32 / SCALE,
```
Comment (Herschel) [inline core/src/backend/audio.rs:572]: Weird behavior!

I think the multiply should be integer using `wrapping_mul`, something like:
`left_to_left: ((other.left_to_left.wrapping_mul(other.volume) << 2) >> 2) as f32 / SCALE,`
Here's an example that this fixes, but not 100% sure. 512 * 4096 = 2^32 -> wraps to 0
```
var s = new Sound(mc);
s.attachSound("marching");

var t = {
	ll: 4096,
	lr: 0,
	rl: 0,
	rr: 0
};
s.setVolume(1048576);
s.setTransform(t);
var t = s.getTransform();
trace("ll: " + t.ll);
trace("lr: " + t.lr);
trace("rl: " + t.rl);
trace("rr: " + t.rr);
trace(s.getVolume());
s.start();

```

[sound_vol_wrap.zip](https://github.com/ruffle-rs/ruffle/files/7117848/sound_vol_wrap.zip)

**Review by Herschel — COMMENTED** [review]
Such weird 
Thank you!

**Pull request merged**